prime-router/src/test/kotlin/azure/ReportFunctionTests.kt

```diff
@@ -1201,4 +1206,134 @@ class ReportFunctionTests {
+
+        private fun setupReportFunction(ackEnabled: Boolean): ReportFunction {
+            val mockEngine = mockk<WorkflowEngine>()
+            val mockActionHistory = mockk<ActionHistory>(relaxed = true)
+            val mockReportStreamEventService = mockk<ReportStreamEventService>(relaxed = true)
+            val mockSettings = mockk<SettingsProvider>()
+            val mockReceiver = mockk<UniversalPipelineReceiver>()
+            val mockAction = mockk<Action>()
+            val mockDb = mockk<DatabaseAccess>()
+            mockkObject(BlobAccess.Companion)
+            mockkObject(SubmissionReceiver.Companion)
+
+            val sender = UniversalPipelineSender(
+                name = "Test Sender",
+                organizationName = "org",
+                format = MimeFormat.HL7,
+                hl7AcknowledgementEnabled = ackEnabled,
+                topic = Topic.FULL_ELR,
+            )
+            val report = Report(
+                Schema(name = "one", topic = Topic.TEST, elements = listOf(Element("a"), Element("b"))), listOf(),
+                sources = listOf(ClientSource("myOrg", "myClient")),
+                metadata = UnitTestUtils.simpleMetadata
+            )
+            val submission = DetailedSubmissionHistory(
+                1,
+                TaskAction.receive,
+                OffsetDateTime.now(),
+                reports = mutableListOf(),
+                logs = emptyList()
+            )
+
+            every { mockEngine.settings } returns mockSettings
+            every { mockSettings.findSender(any()) } returns sender
+            every { BlobAccess.Companion.getBlobConnection(any()) } returns "testconnection"
+            every { SubmissionReceiver.getSubmissionReceiver(any(), any(), any()) } returns mockReceiver
+            every {
+                mockReceiver.validateAndMoveToProcessing(
+                    any(), any(), any(), any(), any(), any(), any(), any(), any()
+                )
+            } returns report
+            every { mockEngine.recordAction(any()) } just Runs
+            every { mockActionHistory.action } returns mockAction
+            every { mockAction.actionId } returns 5
+            every { mockEngine.db } returns mockDb
+            // I don't agree with ktlint on this one
+            every {
+                mockDb.transactReturning(
+                    any<
+                        (
+                DataAccessTransaction,
+            ) -> DetailedSubmissionHistory?
+                        >()
+                )
+            } returns submission
+
+            return ReportFunction(mockEngine, mockActionHistory, mockReportStreamEventService)
+        }
+
+        @Test
+        fun `return ack if requested and enabled`() {
+            val reportFunction = setupReportFunction(ackEnabled = true)
+
+            val body = """
+            MSH|^~\&|Epic|Hospital|LIMS|StatePHL|20241003000000||ORM^O01^ORM_O01|4AFA57FE-D41D-4631-9500-286AAAF797E4|T|2.5.1|||AL|NE
```

Comment:
Perhaps an additional test with a batched message would be good to validate the following requirement:

`Applies to single message submission only. Batch message submission should return current ReportStream JSON response. `